### ISSUE TITLE
Autocomplete keywords with spaces

### ISSUE THREAD

**Issue by glcraft** [Microsoft/vscode-extension-samples#86]
Note: This is not a issue from samples directly. I don't know where to post this knowing that it's related to vscode extension. Tell me if i'm wrong.

Well, I'm creating a language extension using tmLanguage and LPS sample.  The language I work for use spaces between words. For example `APPEND TO ARRAY` is a method and `Size of array` is a function of this language. Colorisation using tmLanguage looks fine but when I try to autocomplete a function with ^Space, vscode only takes the current word.

![screenshot_59](https://user-images.githubusercontent.com/6421451/44095690-c16c505a-9fd9-11e8-834a-97569c1a0e88.png)

Worse : if I autocomplete this words, it only replaces the current word :

![screenshot_60](https://user-images.githubusercontent.com/6421451/44095800-ffb52242-9fd9-11e8-957b-a380d9b0f29e.png)

So is there a way to autocomplete keywords with space ?

**Comment by Tyriar** [comment]
I think @mjbvz knows about this (he's currently on leave so may take a while to respond).

**Comment by mjbvz** [comment]
@glcraft You need to set the range of the completion: `CompletionItem.range`. It defaults to the word range which won't work properly if the entry contains spaces

**Comment by glcraft** [comment]
Thank you for your answer.

I use the `vscode-languageserver` library which doesn't have `CompletionItem.range`. There's `CompletionItem.textEdit` but redefining all keyword stored in my CompletionItem[] with the range I would replace would not be powerful.

I searched around the client API beginning to `CompletionItem.range` then I found `LanguageConfiguration.wordPattern`, which define what a word is compose to. When I changed it to `/[\w ]+/`, completion research work well by selection the whole keyword including space. That's exactly what I needed.

Thank a lot for helping.
